src/wp-includes/taxonomy.php

```diff
@@ -1531,65 +1531,77 @@ function unregister_term_meta( $taxonomy, $meta_key ) {
  *               Returns 0 if term ID 0 is passed to the function.
  */
 function term_exists( $term, $taxonomy = '', $parent = null ) {
-	global $wpdb;
+	global $_wp_suspend_cache_invalidation;
 
 	if ( null === $term ) {
 		return null;
 	}
 
-	$select     = "SELECT term_id FROM $wpdb->terms as t WHERE ";
-	$tax_select = "SELECT tt.term_id, tt.term_taxonomy_id FROM $wpdb->terms AS t INNER JOIN $wpdb->term_taxonomy as tt ON tt.term_id = t.term_id WHERE ";
-
 	if ( is_int( $term ) ) {
 		if ( 0 === $term ) {
 			return 0;
 		}
-		$where = 't.term_id = %d';
+		$_term = get_term( $term, $taxonomy );
```

Comment:
`get_term()` is cached in `WP_Term::get_instance()`

The `$_wp_suspend_cache_invalidation` logic will need to account for this.

Are you able to set up a tests for each of the input types, a data provider would be best bet.

Reply:
Why is this needed? `$_wp_suspend_cache_invalidation` logic is needed because of caches in get_terms / WP_Term_Query caching. This is not an issue here.

Reply:
`get_term()` also caches so can be incorrect if the cache is suspended.

It's easier to demonstrate with a test so I've pushed 2a13edf9c6a8e64174ec613bdf291a5ffd877d2c to your branch to show what is happening. The test currently fails.

Reply:
@peterwilsoncc Great catch! I have massively refactored this patch to ensure that use case is continues to work.